src/main/java/com/puppycrawl/tools/checkstyle/checks/TrailingCommentCheck.java

```diff
@@ -289,18 +286,9 @@ private void checkBlockComment(DetailAST ast) {
      * @return true if the comment if legal.
      */
     private boolean isLegalBlockComment(DetailAST ast, String comment) {
-        final boolean legal;
-
-        // multi-line comment can not be legal
-        if (legalComment == null
-                || !TokenUtil.areOnSameLine(ast.getFirstChild(), ast.getLastChild())) {
-            legal = false;
-        }
-        else {
-            final String commentText = comment.trim();
-            legal = legalComment.matcher(commentText).find();
-        }
-        return legal;
+        return legalComment != null
+                && TokenUtil.areOnSameLine(ast.getFirstChild(), ast.getLastChild())
+                && legalComment.matcher(comment).find();
```

Comment:
Here we employ similar logic that we do for single line comments, with no trimming of content.